Sources/Components/Broadcast/BroadcastContainerView/BroadcastContainerView.swift

```diff
@@ -0,0 +1,258 @@
+            return CGSize(width: frame.size.width, height: 0)
+        }
+
+        let broadcastViewTotalSpacing = ((CGFloat(contentView.arrangedSubviews.count) * LayoutConstants.BroadcastView.spacing) - .mediumLargeSpacing)
+        let verticalSpacing = LayoutConstants.ContentView.insets.top + broadcastViewTotalSpacing
+        let containerHeight = broadcastViewsTotalHeight + verticalSpacing
+
+        return CGSize(width: frame.size.width, height: containerHeight)
+    }
+}
+
+// MARK: - Private
+
+private extension BroadcastContainerView {
+    func setup() {
+        contentView.frame = CGRect(x: 0, y: 0, width: frame.width, height: 0)
+
+        addSubview(contentView)
+
+        let insets = LayoutConstants.ContentView.insets
+
+        NSLayoutConstraint.activate([
+            contentView.topAnchor.constraint(equalTo: topAnchor, constant: insets.top),
+            contentView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: insets.leading),
+            contentView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: insets.trailing),
+        ])
+
+        contentView.setContentCompressionResistancePriority(.defaultLow, for: .vertical)
+        contentView.setContentHuggingPriority(.defaultHigh, for: .vertical)
+    }
+
+    func layoutBroadcastViews(from broadcasts: [Broadcast]) {
+        let broadcastViews = broadcasts.map { broadcastView(from: $0) }
+
+        broadcastViews.forEach { add($0, to: contentView) }
+    }
+
+    func broadcastView(from broadcast: Broadcast) -> BroadcastView {
+        let broadcastView = BroadcastView(frame: .zero)
+
+        // Set the message on the broadcast view and prepare it in its expanded state
+        broadcastView.present(message: broadcast.message, animated: false)
+
+        return broadcastView
+    }
+
+    func add(_ broadcastView: BroadcastView, to stackView: UIStackView) {
+        broadcastView.translatesAutoresizingMaskIntoConstraints = false
+        broadcastView.isHidden = true
+
+        stackView.addArrangedSubview(broadcastView)
+
+        NSLayoutConstraint.activate([
+            broadcastView.leadingAnchor.constraint(equalTo: stackView.leadingAnchor),
+            broadcastView.widthAnchor.constraint(equalTo: stackView.widthAnchor),
+        ])
+
+        broadcastView.setContentCompressionResistancePriority(.defaultLow, for: .vertical)
+        broadcastView.setContentHuggingPriority(.defaultHigh, for: .vertical)
+
+        let tapRecogninzer = UITapGestureRecognizer(target: self, action: #selector(broadcastViewTapped(_:)))
+        broadcastView.addGestureRecognizer(tapRecogninzer)
+
+        broadcastView.layoutIfNeeded()
+    }
+
+    func remove(_ broadcastView: BroadcastView) {
+        guard let subView = contentView.arrangedSubviews.filter({ $0 == broadcastView }).first else {
+            return
+        }
+
+        contentView.removeArrangedSubview(subView)
+        subView.removeFromSuperview()
+    }
+
+    func reset() {
```

Comment:
First time I read this I thought it was resetting some data, might be good to rename it to `removeSubviews` or something that reflects more what it's doing.

Reply:
how about something like`clearContainer` or `removeBroadcastViews`?

Reply:
Maybe `removeAllBroadcastViews` then is also is consistent with the other `remove` method.